### DISCUSSION TITLE
Collaboration - how to retrieve document text from Ydoc

### DISCUSSION THREAD

**Discussion by JoeRoddy** [ueberdosis/tiptap#3343] [General Questions] [answered]
How do you get the unformatted text from a ydoc instance (not the editor instance)?

From the Yjs docs, it should be `ydoc.getText('text-type');` but I have no idea what the type string is set as for tiptap under the hood (or even if tiptap uses a Text doc for storing data or not).  I'm looking for the equivalent of retrieving the text via `editor.getText()` but in this context I only have access to the `ydoc`.

**Comment by yan-yanishevsky** [comment, marked as answer]
It uses `XmlFragment` actually. I had no success to extract text from it, so I used `yDocToProsemirrorJSON` function from `y-prosemirror` package as follows:

```js
import { yDocToProsemirrorJSON } from 'y-prosemirror'

export function getTextFromYDoc(doc) {
  const json = yDocToProsemirrorJSON(doc, 'default')
  const output = []
  const parseJson = (obj) => {
    if (obj.text) {
      output.push(obj.text)
    } else if (obj.content) {
      obj.content.forEach(parseJson)
    }
  }

  parseJson(json)

  return output.join(' ')
}
```

**Reply by JoeRoddy** [reply]
Beautiful, I can't remember how I solved this, but I'm sure it wasn't as elegant as this, haha
